src/RelayClient.ts

```diff
@@ -424,6 +420,147 @@ class RelayClient extends EnvelopingEventEmitter {
       estimatedGasCorrectionFactor
     );
   }
+
+  public async relayTransaction(
+    envelopingRequest: UserDefinedEnvelopingRequest,
+    requestConfig: RequestConfig
+  ): Promise<Transaction> {
+    const httpClient = new HttpClient();
+    const envelopingRequestDetails = await this._getEnvelopingRequestDetails(envelopingRequest, requestConfig);
+    let activeRelay = await selectNextRelay(httpClient);
+
+    while (activeRelay) {
+      const envelopingTx = await this._prepareHttpRequest(activeRelay.hubInfo, envelopingRequestDetails);
+
+      if (await this._verifyEnvelopingRequest(activeRelay.hubInfo, envelopingTx)) {
+
+        const transaction = await this._attemptRelayTransaction(activeRelay, envelopingTx, httpClient)
+       
+        if (transaction) {
+          return transaction;
+        }
+      }
+
+      activeRelay = await selectNextRelay(httpClient);
+    }
+
+    throw Error(NOT_RELAYED_TRANSACTION);
+  }
+
+  private async _attemptRelayTransaction(
+    { relayInfo: { url }, hubInfo }: RelayInfo,
+    envelopingTx: EnvelopingTxRequest,
+    httpClient: HttpClient
```

Comment:
should I pass the client around, create a new instance whenever I need it, or create an instance during the constructor?

Reply:
I would create an instance on construction

Reply:
IMHO I'd prefer to use the already created instance instead of creating a new instance for each method (especially if the method is a private one)

Reply:
updated

Reply:
I was referring to using the instance passed in the method, but since it's a class, this approach is fine also. We can consider changing it once we decide to go for a more functional approach 👍🏽